tests/testDipoleFitter.py

```diff
@@ -0,0 +1,331 @@
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the LSST License Statement and
+# the GNU General Public License along with this program.  If not,
+# see <https://www.lsstcorp.org/LegalNotices/>.
+
+"""Tests of the DipoelFitAlgorithm and its related tasks and plugins.
+Each test generates a fake image with two synthetic dipoles as input data.
+
+DipoleFitTest.testDipoleAlgorithm - tests the DipoleFitAlgorithm directly and ensures that the recovered
+   fluxes and centroids are close to the input values.
+DipoleFitTest.testDipoleTask - tests the DipoleFitTask on identical synthetic data, after running
+   dipole detection on the image.
+DipoleFitTest.testDipoleEdge - ensures correct handling of dipoles too close to the edge of the image.
+"""
+
+import unittest
+
+import numpy as np
+
+import lsst.utils.tests
+import lsst.afw.table as afwTable
+import lsst.meas.base as measBase
+from lsst.ip.diffim.dipoleFitTask import (DipoleFitAlgorithm, DipoleFitTask, DipoleFitPluginConfig)
+import lsst.ip.diffim.utils as ipUtils
+
+
+class DipoleFitTestGlobalParams(object):
+    """Class to initialize and store global parameters used by all tests below.
+
+    @var display: Display (plot) the output dipole thumbnails (matplotlib)
+    @var verbose: be verbose during fitting
+    @var xc: x coordinate (pixels) of center(s) of input dipole(s)
+    @var yc: y coordinate (pixels) of center(s) of input dipole(s)
+    @var flux: flux(es) of input dipole(s)
+    @var gradientParams: tuple with three parameters for linear background gradient
+    @var offsets: pixel coordinates between lobes of dipoles
+    """
+
+    def __init__(self, xc=None, yc=None):
+        """Store the parameters, create the test image and run detection on it.
+
+        @param xc iterable x coordinate (pixels) of center(s) of input dipole(s)
+        @param yc iterable y coordinate (pixels) of center(s) of input dipole(s)
+        """
+        np.random.seed(666)
+        self.display = False  # Display (plot) the output dipole thumbnails (matplotlib)
+        self.verbose = False  # be verbose during fitting
+
+        self.xc = xc if xc is not None else [65.3, 24.2]
+        self.yc = yc if yc is not None else [38.6, 78.5]
+        self.flux = [2500., 2345.]  # fluxes of pos/neg lobes
+        self.gradientParams = [10., 3., 5.]  # three parameters for linear background gradient
+
+        self.offsets = np.array([-2., 2.])  # pixel coord offsets between lobes of dipoles
+
+        # The default tolerance for comparisons of fitted parameters with input values.
+        # Given the noise in the input images (default noise value of 2.), this is a
+        # useful test of algorithm robustness, and will guard against future regressions.
+        self.rtol = 0.01
```

Comment:
Good idea - I will make `flux`, `offsets` and `gradientParams` into arguments since they affect the image that is generated. `rtol` is not used until after `__init__()` is called, so I will leave that out for now.